src/lib.rs

```diff
@@ -60,6 +60,13 @@ pub struct Chunk<B: IntoBuf> {
 
 // ===== impl Body =====
 
+impl<B: IntoBuf> Body<B> {
+    pub fn is_empty(&self) -> bool {
+        // If the recv side is closed and the receive queue is empty, the body is empty.
+        self.inner.is_recv_eos()
```

Comment:
As per the comment thread below, this will return `false` if the body is consumed and the next pending frame is a trailer.